Civi/RemoteTools/Api4/Action/AbstractRemoteGetFieldsAction.php

```diff
@@ -20,19 +20,39 @@
 namespace Civi\RemoteTools\Api4\Action;
 
 use Civi\Api4\Generic\BasicGetFieldsAction;
-use Civi\RemoteTools\Api4\Action\Traits\ActionHandlerRunTrait;
+use Civi\Api4\Generic\Result;
+use Civi\RemoteTools\Api4\Action\Traits\ActionHandlerTrait;
 use Civi\RemoteTools\Api4\Action\Traits\RemoteContactIdParameterOptionalTrait;
 use Civi\RemoteTools\Api4\Action\Traits\ResolvedContactIdOptionalTrait;
+use Civi\RemoteTools\Api4\Util\WhereUtil;
 
 /**
  * @api
  */
-abstract class AbstractRemoteGetFieldsAction extends BasicGetFieldsAction {
+abstract class AbstractRemoteGetFieldsAction extends BasicGetFieldsAction implements RemoteActionInterface {
 
-  use ActionHandlerRunTrait;
+  use ActionHandlerTrait;
 
   use RemoteContactIdParameterOptionalTrait;
 
   use ResolvedContactIdOptionalTrait;
 
+  /**
+   * @phpstan-return array<int|string, mixed>
+   *
+   * @throws \Civi\RemoteTools\Exception\ActionHandlerNotFoundException
+   */
+  protected function getRecords(): array {
+    // Ensure required for the filtering done by BasicGetFieldsAction are
+    // available.
+    $originalSelect = $this->select;
+    if ([] !== $this->select) {
+      $this->select = array_unique(array_merge($this->select, WhereUtil::getFields($this->where)));
+    }
+    $result = $this->getHandlerResult();
+    $this->select = $originalSelect;
```

Comment:
`$this->select` has `protected` visibility but is not marked as `@api`, so maybe it's safer to use the getter …?

Reply:
Now getters/setters are used.